### DISCUSSION TITLE
推数据到Elasticsearch, 报AlarmType:AGGREGATOR_ADD_ALARM  error:loggroup queue is full

### DISCUSSION THREAD

**Discussion by one-code** [alibaba/ilogtail#982] [Help] [answered]
ES集群为3三个实例，采集的日志为每分钟大概7W条，此时报AlarmType:AGGREGATOR_ADD_ALARM  error:loggroup queue is full。但在同一个logtail中配置的另一个日志文件采集，日志量较少，是没有问题的。

**Comment by one-code** [comment]
使用的是ilogtail-1.6.0, es8.1

**Comment by messixukejia** [comment]
可以在历史discussion搜下loggroup queue is full，有排查方法。下游的可能性比较大。

**Reply by one-code** [reply]
目前看不是下游的问题，下游服务器性能没有太多的波动，感觉是ilogtail发送日志的速度过慢，目前社区版有设置发送日志速度控制的参数吗

**Reply by liangry** [reply]
这个量有点大，原先的 ElasticSearch Flusher 是逐条写，会不会有影响？

**Reply by one-code** [reply]
不是并发写吗，难怪看数据写入1秒也就100+条，ES端看不到写入瓶颈。这个是要作为后期的优化点吗

**Reply by liangry** [reply]
是的，目前正在进行中

**Reply by one-code** [reply]
方便透露大概什么时间发布吗

**Reply by one-code** [reply]
想问下，http的方式，目前也是逐条写入吗

**Comment by liangry** [comment, marked as answer]
> 方便透露大概什么时间发布吗

已经提了 PR，正在审议中，如果想尝鲜，可以看看：https://github.com/liangry/ilogtail/tree/feat-flusher-es
关于功能介绍和讨论：https://github.com/alibaba/ilogtail/pull/979
